src/modules/simulation/simulator_sih/sih.cpp

```diff
@@ -422,6 +426,72 @@ void Sih::generate_ts_aerodynamics()
 	_Ma_B = _R_S2B * Ma_ts - _KDW * _w_B; 	// aerodynamic moments
 }
 
+void Sih::generate_rover_ackermann_dynamics(const float throttle_cmd, const float steering_cmd, const float dt)
+{
+	// --- Constants ---
+	static constexpr float MAX_FORCE = 400.0f;        // [N]
+	const float MAX_STEER_ANGLE = radians(30.f); // [rad]
+	const float WHEEL_BASE = 0.321;        // [m] Distance between front and rear axle
+	const float C = 500.f;                 // [N/rad] Cornering stiffness
+	const float MU_S = 0.5f;               // [-] Static Coefficient of friction
+	const float MU_K = 0.4f;               // [-] Kinetic Coefficient of friction
+	const float MU_R = 0.3f;               // [-] Rolling Coefficient of friction
+	const float ROLLING_THRESHOLD = 0.05f; // [m/s] Threshold for rolling resistance
+
+	matrix::Vector3f v_B = _q.rotateVectorInverse(_v_N); // Nav -> Body
+
+	// --- Compute inputs ---
+	const float delta = MAX_STEER_ANGLE * steering_cmd; // [rad] Steering angle
+	const float F_x = MAX_FORCE * throttle_cmd;         // [N] Throttle force
+
+	// --- Compute forces and moments ---
+	float F_y = 0.f; // [N] Lateral force
+	float M_z = 0.f; // [Nm] Yaw moment
+
+	if (fabsf(v_B(0)) > ROLLING_THRESHOLD) {
+		// Equations based on the lateral dynamics of the bycicle model introduced in [1]
+		// [1] Sri Anumakonda, Everything you need to know about Self-Driving Cars in <30 minutes
+		// Link: https://srianumakonda.medium.com/everything-you-need-to-know-about-self-driving-in-30-minutes-b38d68bd3427
+		const float a_y = C * delta / _MASS - fabsf(v_B(0)) * _w_B(2)
+				  - 2 * C * v_B(1) / (_MASS * fabsf(v_B(0))); // [m/s^2] Lateral acceleration
+		const float psi_dot_dot = WHEEL_BASE * C * delta / _sih_izz.get()
+					  - C * WHEEL_BASE * WHEEL_BASE  * _w_B(2) / (_sih_izz.get() * fabsf(v_B(0))); // [rad/s^2] Yaw acceleration
+		F_y = _MASS * a_y; // Lateral force [N]
+		M_z = _sih_izz.get() * psi_dot_dot; // [Nm] Yaw moment
+	}
+
+	_T_B = Vector3f(F_x, F_y, 0.f);
+	_Mt_B = Vector3f(0.f, 0.f, M_z);
+
+	// // --- Compute drag/friction forces and moments ---
+	Vector3f F_f = Vector3f(0.f, 0.f, 0.f); // [N] Friction force
+	Vector3f F_a = Vector3f(0.f, 0.f, 0.f); // [N] Aerodynamic force (neglect until rover is rolling)
+
+	if (_v_E.norm() < 0.01f) { // Static friction
```

Comment:
Can you define these 0.01 as a constant (incl. description, so how you got to it).